hummingbot/connector/derivative/kucoin_perpetual/kucoin_perpetual_derivative.py

```diff
@@ -506,32 +506,24 @@ async def _user_stream_event_listener(self):
         """
         async for event_message in self._iter_user_event_queue():
             try:
+                event_type = event_message.get("type")
+                event_subject = event_message.get("subject")
+                execution_data = event_message.get("data")
                 endpoint = web_utils.endpoint_from_message(event_message)
                 payload = web_utils.payload_from_message(event_message)
-
-                if endpoint == CONSTANTS.WS_SUBSCRIPTION_POSITIONS_ENDPOINT_NAME:
+                if type(execution_data) == list:
+                    execution_data = execution_data[0]
+
+                if event_type == "message" and event_subject == CONSTANTS.ORDER_CHANGE_EVENT_TYPE:
+                    self._process_trade_event_message(execution_data)

```

Comment:
Nice refactor, now is needed to improve the method _process_trade_event_message. Check the logic implemented in Kucoin exchange, is more clean. Also the bug that Ralph is referring is addressed there. the name is not price is MatchPrice or something like that

Reply:
There have been further improvements here in my latest check-in, do these address your concerns?